src/cmake-tools.ts

```diff
@@ -759,17 +839,71 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
     throw new Error('Invalid method: ' + method);
   }
 
-  get sourceDir() { return this._cmakeDriver.then(d => d.sourceDir); }
+  get sourceDir() {
+    const drv = this.getCMakeDriverInstance();
+
+    return drv.then(d => {
+      if (!d) {
+        return "";
+      }
+      return d.sourceDir
+    });
+  }
+
+  get mainListFile() {
+    const drv = this.getCMakeDriverInstance();
+
+    return drv.then(d => {
+      if (!d) {
+        return "";
+      }
+      return d.mainListFile
+    });
+  }
+
+  get binaryDir() {
+    const drv = this.getCMakeDriverInstance();
+
+    return drv.then(d => {
+      if (!d) {
+        return "";
+      }
+      return d.binaryDir
+    });
+  }
+
+  get cachePath() {
+    const drv = this.getCMakeDriverInstance();
 
-  get mainListFile() { return this._cmakeDriver.then(d => d.mainListFile); }
+    return drv.then(d => {
+      if (!d) {
+        return "";
```

Comment:
I think we should use `''` for all strings.

Reply:
Done